support-frontend/assets/pages/contributions-landing/components/PaymentMethodSelector.tsx

```diff
@@ -112,231 +104,277 @@ const mapDispatchToProps = {
-												subscriptionToExplainerPart,
-											),
-										)}
-									</div>
-								</>
-							),
-						)}
-					{paymentMethods.map((paymentMethod) => (
-						<Radio
-							id={`paymentMethodSelector-${paymentMethod}`}
-							name="paymentMethodSelector"
-							type="radio"
-							value={paymentMethod}
-							onChange={() => {
-								onPaymentMethodUpdate(paymentMethod, props);
-							}}
-							checked={props.paymentMethod === paymentMethod}
-							label={renderLabelAndLogo(paymentMethod)}
-							cssOverrides={radioCss}
-						/>
-					))}
-					{contributionTypeIsRecurring(props.contributionType) &&
-						props.existingPaymentMethods &&
-						props.existingPaymentMethods.length > 0 &&
-						fullExistingPaymentMethods.length === 0 && (
-							<li className="form__radio-group-item">
-								...or{' '}
-								<a
-									className="reauthenticate-link"
-									href={getReauthenticateUrl()}
-								>
-									re-enter your password
-								</a>{' '}
-								to use one of your existing payment methods.
-							</li>
-						)}
+										re-enter your password
+									</a>{' '}
+									to use one of your existing payment methods.
+								</li>
+							)}
+					</>
 				</RadioGroup>
 			) : (
-				noPaymentMethodsErrorMessage
+				<GeneralErrorMessage
+					classModifiers={['no-valid-payments']}
+					errorHeading="Payment methods are unavailable"
+					errorReason="all_payment_methods_unavailable"
+				/>
 			)}
 		</div>
 	);
 }
 
+// ----- Helper components ----- //
+
+function Legend() {
+	return (
+		<div className="secure-transaction">
+			<legend id="payment_method">
+				<ContributionChoicesHeader>Payment Method</ContributionChoicesHeader>
+			</legend>
+			<SecureTransactionIndicator modifierClasses={['middle']} />
+		</div>
+	);
+}
+
+interface PaymentMethodLabelProps {
+	label: string;
+	logo: JSX.Element;
+	isChecked: boolean;
+}
+
+function PaymentMethodLabel({
+	label,
+	logo,
+	isChecked,
+}: PaymentMethodLabelProps) {
+	return (
+		<div css={styles.labelContainer} data-checked={isChecked.toString()}>
+			<div>{label}</div>
+			{logo}
+		</div>
+	);
+}
+
+interface PaymentMethodLogoProps {
+	paymentMethod: PaymentMethod;
+}
+
+function PaymentMethodLogo({ paymentMethod }: PaymentMethodLogoProps) {
+	switch (paymentMethod) {
+		case PayPal:
+			return <SvgPayPalDs />;
+
+		case DirectDebit:
+		case ExistingDirectDebit:
+			return <SvgDirectDebitSymbolDs />;
+
+		case AmazonPay:
+			return <SvgAmazonPayLogoDs />;
+
+		case Sepa:
+			return <SvgSepa />;
+
+		case Stripe:
+		case ExistingCard:
+		default:
+			return <SvgNewCreditCardDs />;
+	}
+}
+
+// ---- Styles ---- //
+
+const styles = {
+	labelContainer: css`
+		display: flex;
+		width: 100%;
+		margin: 0;
+		justify-content: space-between;
+		align-items: center;
+
+		svg {
+			width: 36px;
+			height: 24px;
+			display: block;
+		}
+
+		&[data-checked='false'] {
+			svg {
+				filter: grayscale(100%);
+			}
+		}
```

Comment:
I've started writting my emotion more like this when we need dynamic behavior: putting `data-*` attributes on the html and then css selectors like this. Previously I would have written something like:

```ts
labelContainer: (isChecked: boolean) => css`
```

But that requires emotion to regenerate and parse the css every time that changes (potentially might be cached though).